File: battleship-backend/src/main/java/be/kdg/int5/battleshipbackend/domain/ShipType.java
package be.kdg.int5.battleshipbackend.domain;

public enum ShipType {
    CARRIER(5),
    BATTLESHIP(4),
    CRUISER(3),
    SUBMARINE(3),
    DESTROYER(2);

    private final int length;

    ShipType(int length) {
        this.length = length;
    }

    public int getLength() {
        return length;
    }
}

File: battleship-frontend/src/model/ShipType.ts

export enum ShipType {
    CARRIER = "CARRIER",
    BATTLESHIP = "BATTLESHIP",
    CRUISER = "CRUISER",
    SUBMARINE = "SUBMARINE",
    DESTROYER = "DESTROYER"
}

const shipLengths: { [key in ShipType]: number } = {
    [ShipType.CARRIER]: 5,
    [ShipType.BATTLESHIP]: 4,
    [ShipType.CRUISER]: 3,
    [ShipType.SUBMARINE]: 3,
    [ShipType.DESTROYER]: 2,
};

export function getShipLength(shipType: ShipType): number {
    return shipLengths[shipType];
}
